packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -436,8 +436,12 @@ function i(
                 // Check that the child vnode is either an element or VStatic
                 if (!isNull(childVnode) && (isVBaseElement(childVnode) || isVStatic(childVnode))) {
                     const { key } = childVnode;
+                    // In @lwc/engine-server the fragment doesn't have a tagName, default to the VM's tagName.
                     const tagName =
-                        childVnode.sel ?? StringToLowerCase.call(childVnode.fragment.tagName);
+                        childVnode.sel ??
+                        StringToLowerCase.call(
+                            childVnode.fragment.tagName ?? vmBeingRendered.tagName
```

Comment:
Can we just use `vmBeingRendered.tagName` everywhere?

Reply:
Ah, I didn't think of that but it works! 😃